src/header/index.tsx

```diff
@@ -0,0 +1,64 @@
+import * as css from '../theme/default/header.m.css';
+import theme from '../middleware/theme';
+import { RenderResult } from '@dojo/framework/core/interfaces';
+import { ThemedProperties } from '@dojo/framework/core/mixins/Themed';
+import { create, tsx } from '@dojo/framework/core/vdom';
+
+export interface HeaderProperties extends ThemedProperties {
+	/** Determines if this header is fixed */
+	sticky?: boolean;
+}
+
+export type HeaderChildren =
+	| {
+			/** Renderer for leading elements like icons */
+			leading?(): RenderResult;
+			/** Renderer for the header title */
+			title?(): RenderResult;
```

Comment:
What do you think about making `title` mandatory and thus no longer allowing `HeaderChildren` to be undefined? I'm not sure that the header makes a lot of sense without any content?